base-template.sh

```diff
@@ -774,61 +787,140 @@ show_prompt() {
         fi
     fi
 
-    # shellcheck disable=SC2181
+    echo "Installing a clone of \`$GITHOOKS_CLONE_URL\` to \`$INSTALL_DIR/release\` ..."
+
+    CLONE_OUTPUT=$(git clone \
+        -c core.hooksPath=/dev/null \
+        --depth 1 \
+        --single-branch \
+        --branch "$GITHOOKS_CLONE_BRANCH" \
+        "$GITHOOKS_CLONE_URL" "$INSTALL_DIR/release" 2>&1)
+
+    #shellcheck disable=2181
     if [ $? -ne 0 ]; then
-        echo "! Cannot download file \`$DOWNLOAD_FILE\` - command failed" >&2
+        echo "! Cloning of repository \`$GITHOOKS_CLONE_URL\`"
+        echo "! to install directory \`$INSTALL_DIR/release\` failed with output: " >&2
+        echo "$CLONE_OUTPUT" >&2
         return 1
     fi
+
+    git config --global githooks.autoupdate.releaseCloneUrl "$GITHOOKS_CLONE_URL"
+    git config --global githooks.autoupdate.releaseCloneBranch "$GITHOOKS_CLONE_BRANCH"
+
     return 0
 }
 
-#####################################################
-# Loads the contents of the latest install
-#   script into a variable.
+############################################################
+# Checks whether the given directory
+#   is a Git repository (bare included) or not.
 #
-# Sets the ${INSTALL_SCRIPT} variable
+# Returns:
+#   1 if failed, 0 otherwise
+############################################################
+is_git_repo() {
+    git -C "$1" rev-parse >/dev/null 2>&1 || return 1
+}
+
+#####################################################
+# Updates the release clone in the install folder.
 #
 # Returns:
-#   1 if failed the load the script, 0 otherwise
+#   1 if failed, 0 otherwise
 #####################################################
-fetch_latest_update_script() {
-    echo "^ Checking for updates ..."
+update_release_clone() {
 
-    INSTALL_SCRIPT="$(mktemp)"
-    if ! download_file "install.sh" "$INSTALL_SCRIPT"; then
-        echo "! Failed to check for updates" >&2
-        return 1
+    CLONE_DIR="$INSTALL_DIR/release"
+    PULL_ONLY="true"
+
+    # We do a fresh clone if:
+    # - we have an existing clone AND
+    #   - url or branch of the existing clone does not match the settings OR
+    #   - the existing clone has modifications (should not be the case)
+    # - we dont have an existing clone
+
+    GITHOOKS_CLONE_URL=$(git config --global githooks.autoupdate.releaseCloneUrl)
+    GITHOOKS_CLONE_BRANCH=$(git config --global githooks.autoupdate.releaseCloneBranch)
+
+    if is_git_repo "$CLONE_DIR"; then
+        URL=$(git -C "$CLONE_DIR" config remote.origin.url)
+        BRANCH=$(git -C "$CLONE_DIR" symbolic-ref -q --short HEAD)
+
+        if [ "$URL" != "$GITHOOKS_CLONE_URL" ] ||
+            [ "$BRANCH" != "$GITHOOKS_CLONE_BRANCH" ] ||
+            git -C "$CLONE_DIR" diff --quiet; then
+            PULL_ONLY="false"
+        fi
+    else
+        PULL_ONLY="false"
     fi
+
+    if [ "$PULL_ONLY" = "true" ]; then
+        PULL_OUTPUT=$(
+            git -C "$CLONE_DIR" \
+                --work-tree="$CLONE_DIR" \
```

Comment:
Remind me again, why do we need so many flags on this pull?
Is it to stop the git variables from the current hook invocations leak into this other git invocation?

Reply:
Correct. With git clone this is not so much a problem. But we had some strange errors because of this, better we guard this with a few extra options (unfortunately)